Sources/Basics/HTTPClient.swift

```diff
@@ -33,14 +33,16 @@ public struct HTTPClient {
     public typealias Handler = (Request, @escaping (Result<Response, Error>) -> Void) -> Void
 
     public var configuration: HTTPClientConfiguration
+    private let diagnosticsEngine: DiagnosticsEngine
     private let underlying: Handler
 
     // static to share across instances of the http client
     private static var hostsErrorsLock = Lock()
     private static var hostsErrors = [String: [Date]]()
 
-    public init(configuration: HTTPClientConfiguration = .init(), handler: Handler? = nil) {
+    public init(configuration: HTTPClientConfiguration = .init(), handler: Handler? = nil, diagnosticsEngine: DiagnosticsEngine = .init()) {
```

Comment:
Would it make more sense conceptually to have a `nil` diagnostics engine if the caller doesn't pass one?  I realize this would mean conditionalizing the call sites, but since the diagnostics won't go anywhere with the default one, I wonder if that it's the right thing.

Reply:
I can go either way. If making it optional is the convention happy to follow that